<launch>
  <arg name="uuv_name" />
  <arg name="joy_id" default="0"/>
  <!-- Joystick mapping - default: XBox 360 controller -->
  <!-- To activate the vehicle teleop, the deadman button has to remain pressed -->
  <arg name="deadman_button" default="-1"/>
  <arg name="exclusion_buttons" default="4,5"/>
  <arg name="axis_roll" default="-1"/>
  <arg name="axis_pitch" default="-1"/>
  <arg name="axis_yaw" default="0"/>
  <arg name="axis_x" default="4"/>
  <arg name="axis_y" default="3"/>
  <arg name="axis_z" default="1"/>
  <arg name="gain_roll" default="0.0"/>
  <arg name="gain_pitch" default="0.0"/>
  <arg name="gain_yaw" default="0.2"/>
  <arg name="gain_x" default="0.3"/>
  <arg name="gain_y" default="0.3"/>
  <arg name="gain_z" default="0.3"/>
  <arg name="output_topic" default="cmd_vel"/>
  <!-- The type of message can also be geometry_msgs/Accel -->
  <arg name="message_type" default="twist"/>

  <group ns="$(arg uuv_name)">
    <node pkg="joy" type="joy_node" name="joystick">
      <param name="autorepeat_rate" value="10" />
      <param name="dev" value="/dev/input/js$(arg joy_id)"/>
    </node>

    <node pkg="uuv_teleop" type="vehicle_teleop.py" name="joy_uuv_velocity_teleop">
      <remap from="output" to="/$(arg uuv_name)/$(arg output_topic)"/>
      <remap from="joy" to="/$(arg uuv_name)/joy"/>
      <rosparam subst_value="true">
        type: $(arg message_type)
        deadman_button: $(arg deadman_button)
        exclusion_buttons: [$(arg exclusion_buttons)]
        mapping:
          x:
            axis: $(arg axis_x)
            gain: $(arg gain_x)
          y:
            axis: $(arg axis_y)
            gain: $(arg gain_y)
          z:
            axis: $(arg axis_z)
            gain: $(arg gain_z)
          roll:
            axis: $(arg axis_roll)
            gain: $(arg gain_roll)
          pitch:
            axis: $(arg axis_pitch)
            gain: $(arg gain_pitch)
          yaw:
            axis: $(arg axis_yaw)
            gain: $(arg gain_yaw)
      </rosparam>
    </node>

  </group>
</launch>
